extensions/common/sql/sql-bootstrapper/src/main/java/org/eclipse/edc/sql/bootstrapper/SqlSchemaBootstrapperImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.sql.bootstrapper;
+
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.sql.QueryExecutor;
+import org.eclipse.edc.transaction.datasource.spi.DataSourceRegistry;
+import org.eclipse.edc.transaction.spi.TransactionContext;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.eclipse.edc.spi.result.Result.failure;
+import static org.eclipse.edc.spi.result.Result.success;
+
+public class SqlSchemaBootstrapperImpl implements SqlSchemaBootstrapper {
+    private final TransactionContext transactionContext;
+    private final QueryExecutor queryExecutor;
+    private final List<QueuedStatementRecord> statements = new ArrayList<>();
+    private final DataSourceRegistry dataSourceRegistry;
+    private final Monitor monitor;
+
+    public SqlSchemaBootstrapperImpl(TransactionContext transactionContext, QueryExecutor queryExecutor, DataSourceRegistry dataSourceRegistry, Monitor monitor) {
+        this.transactionContext = transactionContext;
+        this.queryExecutor = queryExecutor;
+        this.dataSourceRegistry = dataSourceRegistry;
+        this.monitor = monitor;
+    }
+
+
+    @Override
+    public void addStatementFromResource(String datasourceName, String resourceName, ClassLoader classLoader) {
+        try (var sqlStream = classLoader.getResourceAsStream(resourceName)) {
+            var sql = new Scanner(Objects.requireNonNull(sqlStream)).useDelimiter("\\A").next();
+            statements.add(new QueuedStatementRecord(resourceName, datasourceName, sql));
+        } catch (IOException e) {
+            throw new EdcException(e);
+        }
+    }
+
+    public Result<Void> executeSql() {
```

Comment:
this method should be put in the interface (that will save the explicit cast in the `SqlSchemaBootstrapperExtension`)

Reply:
I don't want this method in the interface, for it could entice contributing modules to invoke it, which could potentially cause problems. The cast is ugly, I'll find another way.

Reply:
ok, I understand it, but this necessity is not explicit (and being misinterpreted, e.g. from a future-me :) ), that could be achieved by keeping only the `addStatement` method in the bootstrapper and moving the execution logic to a module-private service, that won't be registered on the runtime, this way the `executeSql` will effectively become not callable from other modules.

Reply:
thats a lot of ceremony for one single method. if `executeSql()` is package-private and documented, it should be clear enough what is done. 
In addition, I didn't put the interface in an SPI package on purpose, because it is not intended as an extension point anyway.

Reply:
I tried it, and it looks OK, but I'm not really sure splitting such a trivial task in two classes (and two test classes) is really worth it TBH.